src/SideWindow.ts

```diff
@@ -3,26 +3,47 @@ import * as ko from "knockout";
 import { Eagle } from './Eagle';
 import { Utils } from './Utils';
 import { Setting } from "./Setting";
+import { UiModeSystem } from "./UiModes";
 
 export class SideWindow {
-    shown : ko.Observable<boolean>;
```

Comment:
At first I wasn't sure why you had removed shown but kept width. Even though both have been moved to settings.

Then I remembered our discussion about how width changes at high frequency, for example when dragging the window.

Maybe just put a quick comment next to width, explaining why it remains, why it requires a different pattern than 'shown'.